packages/frames.js/src/core/middleware/parseFramesMessage.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  FrameActionPayload,
+  FrameMessageReturnType,
+  getFrameMessage,
+} from "../..";
+import { FramesMiddleware } from "../types";
+
+class RequestBodyNotJSONError extends Error {
+  constructor() {
+    super("Invalid frame action payload");
+  }
+}
+
+class InvalidFrameActionPayloadError extends Error {
+  constructor() {
+    super("Invalid frame action payload");
+  }
+}
+
+function isValidFrameActionPayload(value: any): value is FrameActionPayload {
+  return (
+    value &&
+    typeof value === "object" &&
+    "trustedData" in value &&
+    "untrustedData" in value
+  );
+}
+
+async function decodeFrameActionPayloadFromRequest(
+  request: Request
+): Promise<FrameActionPayload | undefined> {
+  try {
+    // use clone just in case someone wants to read body somewhere along the way
+    const body = await request
+      .clone()
+      .json()
+      .catch(() => {
+        throw new RequestBodyNotJSONError();
+      });
+
+    if (!isValidFrameActionPayload(body)) {
+      throw new InvalidFrameActionPayloadError();
+    }
+
+    return body;
+  } catch (e) {
+    if (
+      e instanceof RequestBodyNotJSONError ||
+      e instanceof InvalidFrameActionPayloadError
+    ) {
+      return undefined;
+    }
+
+    console.error(e);
+
+    return undefined;
+  }
+}
+
+type FramesMessageContext = {
+  message?: FrameMessageReturnType<{ fetchHubContext: false }>;
+};
+
+export function parseFramesMessage(): FramesMiddleware<FramesMessageContext> {
```

Comment:
@stephancill this middleware is just basic message parsing based on our `getFrameMessage` could you please take care of this middleware if there is something missing?

Also based on your idea to support multiple specs I think that this middleware automatically allows you to use different protocol by calling next with no context when the message is not present or is invalid.